addons/wallet/models/gift_card.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+
+import re
+from secrets import token_hex
+
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class GiftCard(models.Model):
+    # inherited Fields
+    _name = "gift.card"
+    _description = "gift card model"
+    _order = 'id desc'
+    _sql_constraints = [
+        ('unique_gift_card_code', 'UNIQUE(code)',
+         'The gift card code must be unique.'),
+        ('check_initial_amount', 'CHECK(initial_amount > 0)',
+         'The initial amount must be strictly positive.')
+    ]
+
```

Comment:
Since I don't see any name field... what will be the display_name ?
Not really "Clean" ^^
![image](https://user-images.githubusercontent.com/4848797/105706947-7dd39380-5f12-11eb-9830-6b256968faff.png)

Reply:
so i should add  a name field ?